### PR TITLE
Adding registering actor from Type but not only GenericType

### PR DESCRIPTION
**Pull request by nero-philip-wang**
# Description

If some types are found by reflection in runtime, I cannot register them. So I add registering actor from Type but not only GenericType so that we can register actor automatically.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #202  

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs [hunk 1]
```diff
@@ -46,6 +46,17 @@ public void TestExplicitActorType()
             Assert.Contains(RenamedActorTypeName, actorRuntime.RegisteredActorTypes, StringComparer.InvariantCulture);
         }
 
+        [Fact]
+        public void TestRegisterOtherTypeAtRuntime()
+        {
+            var actorType = typeof(ActorRuntimeTests);
+            var actorRuntime = new ActorRuntime();
+
+            actorRuntime.RegisterActor(actorType);
```
Comment (amanbha) [inline test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs]: Can we check that it throws InvalidArgumentException as ActorRuntimeTests is not deriving from Actor?

File: test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs [hunk 2]
```diff
@@ -46,6 +46,17 @@ public void TestExplicitActorType()
...
+
+            Assert.DoesNotContain(actorType.Name, actorRuntime.RegisteredActorTypes, StringComparer.InvariantCulture);
+        }
+
```
Comment (amanbha) [inline test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs]: Add another test verifying the functionality which is being added.

Comment (LMWF) [reply test/Dapr.Actors.Test/Runtime/ActorRuntimeTests.cs]: +1 Please add a case where it's successful.

**Comment by amanbha** [comment]
@nero-philip-wang Could you please update the PR to address review comments so that it can be merged?

**Comment by amanbha** [comment]
@nero-philip-wang any updates on the PR?

**Comment by amanbha** [comment]
@nero-philip-wang closing it due to lack of response, please feel free to reopen when its ready to be merged.

**Pull request closed without merging**